unity-renderer/Assets/Scripts/MainScripts/DCL/WorldRuntime/SceneBoundariesController/SceneBoundsChecker.cs

```diff
@@ -287,9 +285,12 @@ protected void UpdateEntityCollidersValidState(MeshesInfo meshesInfo, bool isIns
 
         protected void UpdateComponents(IDCLEntity entity, bool isInsideBoundaries)
         {
-            IOutOfSceneBoundariesHandler[] components = entity.gameObject.GetComponentsInChildren<IOutOfSceneBoundariesHandler>();
+            if(!DataStore.i.sceneBoundariesChecker.componentsCheckSceneBoundaries.ContainsKey(entity.entityId))
+                return;
+            
+            List<IOutOfSceneBoundariesHandler> components = DataStore.i.sceneBoundariesChecker.componentsCheckSceneBoundaries[entity.entityId];
```

Comment:
Brian's (RIP) idea was to make components agnostic to SceneBoundariesChecker. It should be an independant system that would gather anything needed.

That being said... I dont think there's an easy way to achieve it now that components are no longer monobehaviours.

Maybe in the lifecycle of adding a component an event could be raised somewhere so the SceneBoundariesChecker silently listen?

If we could achieve that, the `SceneBoundariesChecker` could become a plugin and be decoupled.